src/irr/asset/CGLSLBRDFBuiltinIncludeLoader.h

```diff
@@ -334,6 +334,8 @@ vec3 irr_glsl_ashikhmin_shirley_cos_eval(in irr_glsl_BSDFAnisotropicParams param
 R"(#ifndef _IRR_BSDF_BRDF_SPECULAR_BECKMANN_INCLUDED_
 #define _IRR_BSDF_BRDF_SPECULAR_BECKMANN_INCLUDED_
 
+#include <irr/builtin/glsl/bsdf/common.glsl>
+
```

Comment:
couple pointers:
- start with implementing sampling for lambertian
- `*cos_gen_sample` should take the ViewSurfaceInteraction as an `in` parameter (we need to know View, Normal, NdotV and possibly later the ray derivatives or ray covaraiance matrices)
- do an overload for `*cos_gen_sample` that takes `vec2` and `uvec2` then we can implement

Reply:
Since `irr_glsl_ViewSurfaceInteraction` does not contain normalized `V` nor `N` we should introduce a `irr_glsl_NormalizedViewSurfaceInteraction` which contains a normalized `V`, `N` and replace a few functions (`bsdf_cos_sample`, `irr_glsl_calcBSDFIsotropicParams` etc. as well as change the type inside the `irr_glsl_BSDFIsotropicParams` struct) that take `irr_glsl_ViewSurfaceInteraction` with ones that take `irr_glsl_NormalizedViewSurfaceInteraction` instead.

Reply:
hmm we could call `irr_glsl_NormalizedViewSurfaceInteraction` as `irr_glsl_IsotropicViewSurfaceInteraction` instead

Reply:
btw for lambert, etc. 

I want concentric mapping not radial
http://www.pbr-book.org/3ed-2018/Monte_Carlo_Integration/2D_Sampling_with_Multidimensional_Transformations.html#SamplingaUnitDisk

also to be packaged off to a function `vec2 irr_glsl_concentricMapping(in vec2 sample)` in some header with Stratified Map and Sample in the name